lib/util/visit-require.js

```diff
@@ -54,11 +59,16 @@ module.exports = function visitRequire(
             })
 
             for (const { node } of references) {
-                const targetNode = node.arguments[0]
+                const targetNode =
+                    /** @type {import('estree').CallExpression} */ (node)
```

Comment:
Why is it necessary to cast the value here? Could that be added in the comment maybe?

Reply:
This is because `arguments` did not exist on `RichNode`.
I could not think of a good way to mark it as a `CallExpression` other than casting
![image](https://github.com/eslint-community/eslint-plugin-n/assets/2230835/dabde81e-7e15-4fef-b270-04dfa2c2d9e6)

Reply:
This solves it, since the union is [discriminated](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#discriminated-unions) on the `type` property:

```
                if (node.type !== 'CallExpression') {
                    continue
                }

                const targetNode = node.arguments[0]
```